mlflow/store/tracking/file_store.py

```diff
@@ -316,6 +317,16 @@ def delete_run(self, run_id):
         new_info = run_info._copy_with_overrides(lifecycle_stage=LifecycleStage.DELETED)
         self._overwrite_run_info(new_info)
 
+    def _hard_delete_run(self, run_id):
+        _, run_dir = self._find_run_root(run_id)
+        os.remove(os.path.join(run_dir, FileStore.META_DATA_FILE_NAME))
+        shutil.rmtree(os.path.join(run_dir, FileStore.METRICS_FOLDER_NAME))
+        shutil.rmtree(os.path.join(run_dir, FileStore.PARAMS_FOLDER_NAME))
+        shutil.rmtree(os.path.join(run_dir, FileStore.TAGS_FOLDER_NAME))
+
+    def _get_deleted_runs(self):
```

Comment:
Based on my comment above, I think we can remove this method and rely on the Search API instead.

Reply:
See my answer to the first comment about this.

Reply:
@t-henri Your previous comment makes sense to me. Can we implement this method for the FileStore via the search API (e.g., search with the default max_runs limit and `viewtype = 'deleted'`), since this seems to be the only thing preventing `mlflow gc` from working against the file store?

Reply:
Done